### PR TITLE
Modernize things

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,4 @@
+* text=auto eol=lf
+
+*.bat text eol=crlf
+*.jar binary
```
Comment (saket) [inline .gitattributes:4]: What do these options do?

File: .github/workflows/test_and_release_snapshot.yml [hunk 2]
```diff
@@ -19,17 +19,19 @@ jobs:
...
           distribution: 'zulu'
           java-version: 17
-      - uses: gradle/gradle-build-action@v2.4.2
+      - uses: gradle/gradle-build-action@v2
+        with:
+          gradle-home-cache-cleanup: true
 
       - name: Run unit tests
-        run: ./gradlew check
+        run: ./gradlew build
```
Comment (saket) [inline .github/workflows/test_and_release_snapshot.yml:27]: Good idea!

File: library/build.gradle.kts [hunk 3]
```diff
@@ -10,13 +9,9 @@ dependencies {
...
 }
 
-java {
-  toolchain.languageVersion.set(JavaLanguageVersion.of(11))
-}
-
 // Used on CI to prevent publishing of non-snapshot versions.
 tasks.register("throwIfVersionIsNotSnapshot") {
-  val libraryVersion = properties["VERSION_NAME"] as String
+  val libraryVersion = providers.gradleProperty("VERSION_NAME").orNull.orEmpty()
```
Comment (saket) [inline library/build.gradle.kts:21]: TIL

File: library/src/main/kotlin/me/saket/filesize/FileSize.kt [hunk 4]
```diff
@@ -49,7 +49,8 @@ class FileSize(private val bytes: Long) : Comparable<FileSize> {
       is Byte,
       is Short,
       is Int,
-      is Long -> Math.multiplyExact(bytes, other.toLong())
+      is Long,
+      -> Math.multiplyExact(bytes, other.toLong())
```
Comment (saket) [inline library/src/main/kotlin/me/saket/filesize/FileSize.kt]: Oof this looks terrible. Considering that this is a tiny project with just two source files, I'm okay with not having an auto-formatter. Can you remove this commit that introduces ktlint + spotless from this PR? I'll merge the rest.

File: .gitattributes (hunk 1 above)
Comment (Goooler) [reply .gitattributes:4]: Mark `.bat` files ending with CRLF, it's the default EOL.

File: library/src/main/kotlin/me/saket/filesize/FileSize.kt (hunk 4 above)
Comment (Goooler) [reply library/src/main/kotlin/me/saket/filesize/FileSize.kt]: c926e7aed3eb8881b0f19fc3539e64bd1fdf8427

**Comment by Goooler** [comment]
Please squash this PR once you are ready to merge.

**Comment by saket** [comment]
Thanks! Can you rebase your branch again?

File: library/build.gradle.kts [hunk 5]
```diff
@@ -1,5 +1,4 @@
 plugins {
-  id("java-library")
```
Comment (saket) [inline library/build.gradle.kts:2]: Interesting, is adding this plugin optional?

**Comment by saket** [comment]
Going ahead and merging these changes. I'll fix the compilation errors in a follow-up.

File: library/build.gradle.kts (hunk 5 above)
Comment (Goooler) [reply library/build.gradle.kts:2]: This should be handled by kotlin-jvm and plugin-publish.

**Comment by Goooler** [comment]
I'm on mobile, will fix it later.

File: library/build.gradle.kts (hunk 3 above)
Comment (saket) [reply library/build.gradle.kts:21]: This actually didn't work. `providers.gradleProperty("VERSION_NAME").orNull.orEmpty()` returns an empty string. Know what might be up?

Comment (Goooler) [reply library/build.gradle.kts:21]: I know why, cause you defined the const in sub-project's `gradle.properties`, there's a bug of picking up it in Gradle, see https://github.com/gradle/gradle/issues/23572


https://github.com/saket/file-size/blob/80dd4f2ea411036b7ff8cdf1f9eca165863ccee5/library/gradle.properties#L7

**Pull request merged**